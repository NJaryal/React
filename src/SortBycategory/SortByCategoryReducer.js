import { SET_SORT_BY_RATING, SET_SORT_BY_RELEASEDATE } from './SortByCategoryActions';

function SortByCategoryReducer(state = {
  releaseActive: false,
  ratingActive: true,
}, action) {
  switch (action.type) {
    case SET_SORT_BY_RATING:
      return {
        ...state,
        releaseActive: false,
        ratingActive: true,
      };
    case SET_SORT_BY_RELEASEDATE:
      return {
        ...state,
        releaseActive: true,
        ratingActive: false,
      };
    default:
      return state;
  }
}
export default SortByCategoryReducer;
