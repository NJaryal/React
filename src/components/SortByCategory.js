import React from 'react'

function SortByCategory (props) {
    return (
        <div className="clearfix moviesubheadingSection"> 
            <div className="float-left float-md-left"> <strong>{props.count}</strong> movie found</div>
            <div className="float-right float-md-right">SORT BY                          
                <button type="button" className="btn btn-sm btn-sort_toggle ml-1" onClick={props.sortRelease}>RELEASE DATE</button>
                <button type="button" className="btn btn-sm btn-sort_toggle ml-1" onClick={props.sortRating}>SORTING</button>
            </div>
        </div>
    )
}


export default SortByCategory;
