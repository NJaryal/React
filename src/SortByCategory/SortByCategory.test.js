import React from 'react';
import { shallow } from 'enzyme';
import { SortByCategory } from './SortByCategory';
import { findByTestAttr, checkProps } from '../../Utils';
import SortByCategoryReducer from './SortByCategoryReducer';

const renderSortByCategory = (props = {}) => {
  const component = shallow(<SortByCategory {...props} />);
  return component;
};

describe('SortByCategory component', () => {
  describe('Check PropTypes', () => {
    it('should not throw a warning', () => {
      const expectedprops = {
        count: 10,
        activeSort: 'release',
      };
      const propdError = checkProps(SortByCategory, expectedprops);
      expect(propdError).toBeDefined();
    });
  });

  describe('Have Props', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        count: 10,
        activeSort: 'release',
      };
      wrapper = renderSortByCategory(props);
    });
    it('should render', () => {
      const component = findByTestAttr(wrapper, 'SortByCategory');
      expect(component.length).toBe(1);
    });

    it('should render a count', () => {
      const count = findByTestAttr(wrapper, 'count');
      expect(count.length).toBe(1);
    });

    it('should render a activeSort', () => {
      const activeSort = findByTestAttr(wrapper, 'activeSort');
      expect(activeSort.length).toBe(1);
    });
  });

  describe('reducer', () => {
    it('should return default state', () => {
      const newState = SortByCategoryReducer(undefined, {});
      expect(newState).toEqual({
        activeSort: 'release',
      });
    });
  });

  describe('should show', () => {
    let component;
    beforeEach(() => {
      component = renderSortByCategory();
    });
    it('snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
