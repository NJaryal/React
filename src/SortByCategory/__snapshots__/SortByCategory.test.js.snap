// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortByCategory component should show snapshot 1`] = `
<div
  className="moviesubheadingSection clearfix"
  data-test="SortByCategory"
>
  <div
    className="float-left float-md-left"
    data-test="count"
  >
    <strong />
    movie found
  </div>
  <div
    className="float-right float-md-right"
    data-test="activeSort"
  >
    SORT BY
    <Button
      className="cx(globalStyles['btn-sm'], globalStyles['ml-1']) inactive"
    >
      RELEASE DATE
    </Button>
    <Button
      className="cx(globalStyles['btn-sm'], globalStyles['ml-1']) inactive"
    >
      RATING
    </Button>
  </div>
</div>
`;
